packages/babel-generator/test/fixtures/flow/type-annotations/output.js

```diff
@@ -180,7 +180,7 @@ class Array {
 
 }
 
-var x: () => number | () => string = fn;
+var x: () => number | (() => string) = fn;
```

Comment:
Do you know why the first FunctionTypeAnnotation here is not in parentheses? Isn't its parent also union?

Reply:
It's because it is `() => (number | (() => ... ))`